include/sundials/sundials_iterative.h

```diff
@@ -56,7 +56,7 @@ extern "C" {
  * -----------------------------------------------------------------
  */
 
-enum
+enum SUN_PREC_ID
```

Comment:
To follow the `enum` naming convention, should this be `SUNPrecID` or maybe `SUNPrecType`? I like `Type` better than `ID`.
```suggestion
enum SUNPrecType
```

Reply:
The problem is that it needs to match the typedef type name, so if we did this, we would have to deprecate all of the non-conforming enum names.

Reply:
These were unnamed before so in this case there's nothing to deprecate